Tools/YamlCreate.ps1

```diff
@@ -31,232 +31,663 @@ filter TrimString {
+            $FileExtensions = Read-Host -Prompt 'FileExtensions' | TrimString
+        } while ($FileExtensions.Split(", ").Count -gt '256')
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter any Protocols the package provides a handler for. For example: http, https (Max 16)'
+            $Protocols = Read-Host -Prompt 'Protocols' | TrimString
+        } while ($Protocols.Split(", ").Count -gt '16')
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter any Commands or aliases to run the package. For example: msedge (Max 16)'
+            $Commands = Read-Host -Prompt 'Commands' | TrimString
+        } while ($Commands.Split(", ").Count -gt '16')
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Installer Scope. machine or user'
+            $Scope = Read-Host -Prompt 'Scope' | TrimString
+        } until ([string]::IsNullOrWhiteSpace($Scope) -or ($Scope -eq 'machine' -or $Scope -eq 'user'))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Moniker (friendly name/alias). For example: vscode'
+            $Moniker = Read-Host -Prompt 'Moniker' | TrimString
+        } while ($Moniker.Length -gt 40)
+
+        #        DefaultLocale Manifest      #
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Publisher Url'
+            $PublisherUrl = Read-Host -Prompt 'Publisher Url' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PublisherUrl) -and ($PublisherUrl.Length -lt 5 -or $LicenseUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Publisher Support Url'
+            $PublisherSupportUrl = Read-Host -Prompt 'Publisher Support Url' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PublisherSupportUrl) -and ($PublisherSupportUrl.Length -lt 5 -or $PublisherSupportUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Privacy Url'
+            $PrivacyUrl = Read-Host -Prompt 'Privacy Url' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($PrivacyUrl) -and ($PrivacyUrl.Length -lt 5 -or $PrivacyUrl.Length -gt 2000))
+
+        do {
+            Write-Host ''
+            Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Application Author'
+            $Author = Read-Host -Prompt 'Author' | TrimString
+        } while (-not [string]::IsNullOrWhiteSpace($Author) -and ($Author.Length -lt 1 -or $Author.Length -gt 256))
```

Comment:
Can't this be simplified to `while ($Author.Length -gt 256)` ?

Reply:
This is code from the old YamlCreate, i believe this is in because the values have a a min and max length. (However some min length values seems to have changed)
https://github.com/microsoft/winget-cli/blob/master/schemas/JSON/manifests/v1.0.0/manifest.defaultLocale.1.0.0.json
I can remove all the -lt and IsNullOrWhiteSpace if you'd prefer.

Reply:
I see, so this is just to keep the checks consistently looking, that's fine